internet.go

```diff
@@ -94,59 +106,80 @@ func MacAddress() string {
 	}).(string)
 }
 
-func (internet Internet) domainName() string {
-	return randomString(7, &LangENG) + "." + randomElementFromSliceString(tld)
+func (internet Internet) domainName() (string, error) {
+	domainPart, err := randomString(7, &LangENG)
+	if err != nil {
+		return "", err
+	}
+	return (domainPart + "." + randomElementFromSliceString(tld)), nil
 }
 
 // DomainName generates random domain name
 func (internet Internet) DomainName(v reflect.Value) (interface{}, error) {
-	return internet.domainName(), nil
+	return internet.domainName()
 }
 
 // DomainName get email domain name in string
 func DomainName() string {
 	return singleFakeData(DomainNameTag, func() interface{} {
 		i := Internet{}
-		return i.domainName()
+		d, err := i.domainName()
+		if err != nil {
+			panic(err.Error())
+		}
+		return d
 	}).(string)
 }
 
-func (internet Internet) url() string {
+func (internet Internet) url() (string, error) {
 	format := randomElementFromSliceString(urlFormats)
 	countVerbs := strings.Count(format, "%s")
+	d, err := internet.domainName()
+	if err != nil {
+		return "", nil
+	}
 	if countVerbs == 1 {
-		return fmt.Sprintf(format, internet.domainName())
+		return fmt.Sprintf(format, d), nil
+	}
+	u, err := internet.username()
+	if err != nil {
+		return "", err
 	}
-	return fmt.Sprintf(format, internet.domainName(), internet.username())
+	return fmt.Sprintf(format, d, u), nil
 }
 
 // URL generates random URL standardized in urlFormats const
 func (internet Internet) URL(v reflect.Value) (interface{}, error) {
-	return internet.url(), nil
+	return internet.url()
 }
 
 // URL get Url randomly in string
 func URL() string {
 	return singleFakeData(URLTag, func() interface{} {
 		i := Internet{}
-		return i.url()
+		u, err := i.url()
+		if err != nil {
+			panic(err.Error())
+		}
+		return u
 	}).(string)
 }
 
-func (internet Internet) username() string {
+func (internet Internet) username() (string, error) {
 	return randomString(7, &LangENG)
 }
 
 // UserName generates random username
 func (internet Internet) UserName(v reflect.Value) (interface{}, error) {
-	return internet.username(), nil
+	return internet.username()
 }
 
 // Username get username randomly in string
 func Username() string {
 	return singleFakeData(UserNameTag, func() interface{} {
 		i := Internet{}
-		return i.username()
+		u, _ := i.username()
```

Comment:
This one is not consistent with the others. I see the others is calling panic, but this one is ignored. Let's keep it consistent, either we call panic or ignore it.

Reply:
Fixed!